reactive-banana/src/Reactive/Banana/Internal/Combinators.hs

```diff
@@ -43,30 +43,35 @@ interpret f = Prim.interpret $ \pulse -> runReaderT (g pulse) undefined
 ------------------------------------------------------------------------------}
 -- | Data type representing an event network.
 data EventNetwork = EventNetwork
-    { runStep :: Prim.Step -> IO ()
-    , actuate :: IO ()
-    , pause   :: IO ()
+    { actuated :: IORef Bool
+    , s :: MVar Prim.Network
     }
 
+
+runStep :: EventNetwork -> Prim.Step -> IO ()
+runStep EventNetwork{ actuated, s } f = whenFlag actuated $ do
+    s1 <- takeMVar s                    -- read and take lock
+    -- pollValues <- sequence polls     -- poll mutable data
+    (output, s2) <- f s1                -- calculate new state
+    putMVar s s2                        -- write state
+    output                              -- run IO actions afterwards
+    where
+        whenFlag flag action = readIORef flag >>= \b -> when b action
```

Comment:
```suggestion
    output                              -- run IO actions afterwards
  where
    whenFlag flag action = readIORef flag >>= \b -> when b action
```

I have come to like this style where `where` is only indented 2 spaces instead of the regular 4 spaces. In this way, it becomes clearer that the code following the `where` is "on the same level" as the code before it.